src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -322,27 +308,40 @@ private static bool ValidateName(string name, bool targetingAttributes, ErrorSin
 
             var validName = TryValidateName(
                 name,
-                whitespaceError: Resources.FormatHtmlTargetElementAttribute_NameCannotBeNullOrWhitespace(targetName),
-                characterErrorBuilder: (invalidCharacter) =>
-                    Resources.FormatHtmlTargetElementAttribute_InvalidName(
-                        targetName.ToLower(),
-                        name,
-                        invalidCharacter),
+                targetName,
                 errorSink: errorSink);
 
             return validName;
         }
 
         private static bool TryValidateName(
             string name,
-            string whitespaceError,
-            Func<char, string> characterErrorBuilder,
+            string tagHelperName,
+            ErrorSink errorSink)
+        {
+            return TryValidateName(
+                name,
+                tagHelperName,
+                false,
+                errorSink);
+            
+        }
+
+        private static bool TryValidateName(
```

Comment:
@dougbu  The code actually find all the invalid characters in the name and adds errors for each of the invalid character in the ErrorSink. So, the second may not be applicable.